threescale_api/defaults.py

```diff
@@ -362,13 +379,16 @@ def exists(self, **kwargs) -> bool:
         return self.client.exists(entity_id=self.entity_id, **kwargs)
 
     def delete(self, **kwargs):
-        self.client.delete(entity_id=self.entity_id, **kwargs)
+        self.client.delete(entity_id=self.entity_id, resource=self, **kwargs)
```

Comment:
this 'resource' stuff deserves a discussion/explanation

Reply:
Replied in the answer above.